File: RMIConnector/src/main/java/com/lonelytravelers/rmiconnector/MessageGateway.java
package com.lonelytravelers.rmiconnector;

import com.rabbitmq.client.AMQP.BasicProperties;
import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.DeliverCallback;
import java.io.IOException;
import java.util.concurrent.TimeoutException;
import java.util.logging.Level;
import java.util.logging.Logger;

public class MessageGateway {
    private static ConnectionFactory factory;
    private static Connection requestConnection;
    private static Channel requestChannel;

    static {
        try {
            factory = new ConnectionFactory();
            factory.setHost("rabbitmq");
            requestConnection = factory.newConnection();
            requestChannel = requestConnection.createChannel();
        } catch (IOException | TimeoutException ex) {
            Logger.getLogger(MessageGateway.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    public static void ReceiveRequests() throws IOException {
        requestChannel.exchangeDeclare("connector", "Direct");

        String queueName = requestChannel.queueDeclare().getQueue();
        requestChannel.queueBind(queueName, "connector", "requests");

        DeliverCallback deliverCallback = (consumerTag, delivery) -> {
            var body = delivery.getBody();
            var message = new String(body, "UTF-8");
            CommandRouter.Route(message);
        };

        requestChannel.basicConsume(queueName, true, deliverCallback, consumerTag -> {
        });
    }

    public static void SendAvailableCars(String message) throws IOException {
        requestChannel.exchangeDeclare("cars", "Direct");

        var props = new BasicProperties.Builder().build();

        requestChannel.basicPublish("cars", "available", props, message.getBytes());
    }

}

File: CreationService/MessageGateway.cs
using System;
using System.Text;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;

namespace CreationService
{
    public static class MessageGateway
    {
        private static ConnectionFactory factory;
        private static IConnection connection;
        private static IModel channel;
        static MessageGateway()
        {
            factory = new ConnectionFactory() { HostName = "rabbitmq" };
            connection = factory.CreateConnection();
            channel = connection.CreateModel();
        }

        internal static void ReceiveQueue(string exchangeName, string routingKey, EventHandler<BasicDeliverEventArgs> receivedMethod)
        {
            channel.ExchangeDeclare(exchange: exchangeName, type: ExchangeType.Direct);

            var queueName = channel.QueueDeclare().QueueName;
            channel.QueueBind(queue: queueName,
                              exchange: exchangeName,
                              routingKey: routingKey);

            var consumer = new EventingBasicConsumer(channel);
            consumer.Received += receivedMethod;

            channel.BasicConsume(queue: queueName,
                                 autoAck: true,
                                 consumer: consumer);
        }

        internal static void PublishMessage(string exchangeName, string routingKey, string message)
        {
            channel.ExchangeDeclare(exchange: exchangeName, type: ExchangeType.Direct);

            var body = Encoding.UTF8.GetBytes(message);

            channel.BasicPublish(
                exchange: exchangeName,
                routingKey: routingKey,
                basicProperties: null,
                body: body);
        }
    }
}